bitcoin/src/blockdata/transaction.rs

```diff
@@ -863,7 +863,7 @@ where
         |(count, partial_input_weight, inputs_with_witnesses), prediction| {
             (
                 count + 1,
-                partial_input_weight + prediction.weight().to_wu() as usize,
+                partial_input_weight + prediction.witness_weight().to_wu() as usize,
```

Comment:
I wonder if this could be changed to:

```suggestion
        input_weight += prediction.total_weight().to_wu() as usize;
```

Then `predict_weight_internal` can take `input_weight` instead of `partial_input_weight` and this line can be deleted since the total input weight is already known:

```
    // Lengths of txid, index and sequence: (32, 4, 4).
    // Multiply the lengths by 4 since the fields are all non-witness fields.
    let input_weight = partial_input_weight + input_count * 4 * (32 + 4 + 4);
```

Reply:
Yeah good idea, that's a smart refactor.